emperor/support_files/js/abc-view-controller.js

```diff
@@ -183,5 +183,14 @@ define([
     throw Error('Not implemented');
   };
 
+  /**
+   * Writes the current settings to the active decomposition view(s).
+   * Will be called when the decomposition view is swapped out
+   * for a different view type.
+   */
+  EmperorViewControllerABC.prototype.forceRefresh = function() {
```

Comment:
- With regards to the problem of re-applying settings being a slow operation, perhaps we can instead go marker by marker and copy-over the attributes. Instead of updating the UI and triggering a number of unnecessary calls.